### PR TITLE
Loans: Moment as Seconds & Millis explicitly

### PR DESCRIPTION
**Pull request by lemunozm**
# Description

Fixes: #1360

Instead of having an agnostic `Moment` type, this PR modifies the codebase to reason about `Seconds` or `Millis` and avoid mixing them in pallets.

## Changes and Descriptions

- New trait when the intention is to obtain the seconds, and another to convert into it.
- Added `Seconds` and `Millis` types (by now, there are in `cfg_traits`, but all this machinery should be moved to a future `cfg_utils`: #1380).
- Remove the `Moment` type, which is a too-open name. The name is still used for traits agnostic of the time unit.
- Refactor some pallets to explicitly reason about seconds.
- **Fixed some time bug found in `pallet-loans`**

### Checklist
- [x] `pallet-loans` impl
- [x] `pallet-loans` tests
- [x] `pallet-interest-accrual` impl
- [x] `pallet-pools` impl
- [x] `pallet-pools` tests
- [x] `permission` types => **Because internally it retrived seconds, I modify them to use directly the new `Seconds` type. But I'm not sure if a more correct solution would be to make permission types more independent from secs. Although it would require a refactor to use it from pallet-pool-system correctly I guess. By now, I think using `Seconds` directly simplifies things.**
- [x] `pallet-liquidity-pools` 
- [x] `pallet-registry`
- [x] update runtime configurations with new time types.

### CODE REVIEW

**Review by mustermeiszer — APPROVED** [review]
Do you in the same run change it for all pallets?

**Comment by lemunozm** [comment]
Until I get exhausted 😆

**Comment by lemunozm** [comment]
I think only the following pallets:
- pallet-loans
- pallet-interest-accrual
- pallet-pools

need this change. Other pallet uses moment independently of what `Moment`is, but the above pallets requires `Moment` to be precisely seconds in order to compute correctly the interest.

**Comment by lemunozm** [comment]
This is finished and can be merged

File: libs/traits/src/lib.rs [hunk 1]
```diff
@@ -631,3 +632,32 @@ pub trait ConversionFromAssetBalance<AssetBalance, AssetId, OutBalance> {
...
+impl<T: UnixTime> TimeAsSecs for T {}
+
+/// Trait to convert into seconds
+pub trait IntoSeconds {
+	fn into_seconds(self) -> Seconds;
+}
+
+impl IntoSeconds for Millis {
+	fn into_seconds(self) -> Seconds {
+		self / 1000
```
Comment (mustermeiszer) [inline libs/traits/src/lib.rs:661]: Safe-math?

File: pallets/loans/src/entities/loans.rs [hunk 2]
```diff
@@ -251,15 +249,17 @@ impl<T: Config> ActiveLoan<T> {
...
 			}
 			WriteOffTrigger::PriceOutdated(secs) => match &self.pricing {
-				ActivePricing::External(pricing) => {
-					Ok(now >= pricing.last_updated(pool_id)?.ensure_add(*secs)?)
-				}
+				ActivePricing::External(pricing) => Ok(now
+					>= pricing
+						.last_updated(pool_id)?
+						.into_seconds()
+						.ensure_add(*secs)?),
```
[the remark this answers is not in the document]
Comment (mustermeiszer) [reply pallets/loans/src/entities/loans.rs:262]: What was the bug here? Oracles are actually millies?

[the remark this answers is not in the document]
Comment (mustermeiszer) [reply pallets/loans/src/entities/loans.rs:262]: Assuming we hit this, this means
* `now` is given in seconds
* `last_updated` is given in milli-seconds
* `secs` is given in seconds

The widest false positive in the `Ok(bool)` logic we can get will then be at most `0.999` seconds, and as we are not caring for that in the compatrison we will anyways never hit that, right?

**Review by mustermeiszer — APPROVED** [review]
Thanks! That was tedious.

File: libs/traits/src/lib.rs (hunk 1 above)
Comment (lemunozm) [reply libs/traits/src/lib.rs:661]: A division by `1000` can never crash

File: pallets/loans/src/entities/loans.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (lemunozm) [reply pallets/loans/src/entities/loans.rs:262]: Yes, `orml_oracle` sets the timestamp as milliseconds.

Exactly, so right now, this trigger can not be triggered.

**Comment by lemunozm** [comment]
Thanks for the review!

As a note aside, related to this PR https://github.com/centrifuge/centrifuge-chain/issues/1380, probably Seconds / Millis (and maybe even the traits) would be in `cfg-utils` in the future, to be generic and runtime-unrelated types

**Pull request merged**